src/test/groovy/com/intuit/graphql/authorization/enforcement/AuthorizationTest.groovy

```diff
@@ -0,0 +1,544 @@
+                .query(IntrospectionQuery.INTROSPECTION_QUERY)
+                .context("Test.client4,Test.client2")
+                .build()
+
+        when:
+        def result = graphql.execute(executionInput)
+
+        then:
+        assertThat(result.errors.size()).isEqualTo(0)
+
+        GsonBuilder builder = new GsonBuilder()
+        Gson gson = builder.create()
+        JsonElement res = gson.toJsonTree(result.toSpecification())
+        JsonObject jsonres = res.getAsJsonObject().get("data").getAsJsonObject().get("__schema").getAsJsonObject()
+        assertThat(jsonres.size()).isEqualTo(4)
+        assertThat(jsonres.get("queryType").toString()).isEqualTo("{\"name\":\"Query\"}")
+        assertThat(jsonres.get("mutationType").toString()).isEqualTo("{\"name\":\"Mutation\"}")
+
+        JsonArray types = jsonres.get("types").getAsJsonArray()
+        assertThat(types.size()).isEqualTo(19)
+
+        assertThat(hasValue(types, "kind", "OBJECT", "name", "Author")).isTrue()
+        assertThat(hasValue(types, "kind", "OBJECT", "name", "Book")).isTrue()
+        assertThat(hasValue(types, "kind", "OBJECT", "name", "Query")).isTrue()
+        assertThat(hasValue(types, "kind", "OBJECT", "name", "Mutation")).isTrue()
+        assertThat(hasValue(types, "kind", "INPUT_OBJECT", "name", "BookID")).isTrue()
+        assertThat(hasValue(types, "kind", "INPUT_OBJECT", "name", "BookInput")).isTrue()
+        assertThat(hasValue(types, "kind", "INPUT_OBJECT", "name", "AuthorInput")).isTrue()
+
+        assertThat(CollectionUtils.isEqualCollection(getFields(types, "Query"), Arrays.asList("bookById"))).isTrue()
+        assertThat(CollectionUtils.isEqualCollection(getFields(types, "Author"), Arrays.asList("firstName"))).isTrue()
+        assertThat(CollectionUtils.isEqualCollection(getFields(types, "Book"), Arrays.asList("id", "name", "pageCount", "author"))).isTrue()
+        assertThat(CollectionUtils.isEqualCollection(getFields(types, "Mutation"), Arrays.asList("createNewBookRecord", "updateBookRecord", "removeBookRecord"))).isTrue()
+    }
+
+    private boolean hasValue(JsonArray array, String key1, String value1, String key2, String value2) {
+        for (JsonElement element : array) {
+            JsonObject obj = element.getAsJsonObject()
+            if (obj.get(key1).getAsString().equals(value1) && obj.get(key2).getAsString().equals(value2)) {
+                return true
+            }
+        }
+        return false
+    }
+
+    private List<String> getFields(JsonArray array, String typeName) {
+        for (JsonElement element : array) {
+            JsonObject obj = element.getAsJsonObject()
+            if (obj.get("name").getAsString().equals(typeName)) {
+                JsonArray fields = obj.get("fields").getAsJsonArray()
+                return fields.stream().map { it.getAsJsonObject().get("name").getAsString() }.collect(Collectors.toList())
+            }
+        }
+        return []
+    }
+
+}
```

Comment:
Some comments on groovy style tests 

1. most things should be done inline with multiline strings, like all the files can be kept inline within the test.
2. Groovy tests are less verbose, example maps can be initialized directly
3. assertThat is not needed

Check tests here for example
https://github.com/graph-quilt/graphql-orchestrator-java/tree/master/src/test/groovy/com/intuit/graphql/orchestrator/integration